instrumentation/pg/lib/opentelemetry/instrumentation/pg/patches/connection.rb

```diff
@@ -66,15 +66,15 @@ def lru_cache
           # But, getting that metric in line would force us over the
           # module size limit! We can't win here unless we want to start
           # abstracting things into a million pieces.
-          def span_attrs(kind, *args) # rubocop:disable Metrics/AbcSize
+          def span_attrs(kind, *args) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
             if kind == :query
               operation = extract_operation(args[0])
-              sql = obfuscate_sql(args[0])
+              sql = args[0]
             else
               statement_name = args[0]
 
               if kind == :prepare
-                sql = obfuscate_sql(args[1])
+                sql = args[1]
                 lru_cache[statement_name] = sql
```

Comment:
Trying to think this one through, is the performance better or worse with this change? 

Before we would cache the obfuscated code, but we would obfuscate before we knew that whether or not we would record the attribute for it, so that could have been wasted work.

Now we only obfuscate if the setting is enabled (which means we are recording the attribute), but now we are not caching the obfuscated code so we may obfuscate the same SQL multiple times.  

@ahayworth @fbogsany wdyt?  Is this being a bit too nit picky?

Reply:
I was too zoned in on my changes, I think it makes sense to use LRU cache for obfuscated call

Reply:
I agree - I think the simplest change here is to just replace the original conditionals `if config[:enable_statement_attribute]` and `unless config[:enable_sql_obfuscation]` with (respectively) `unless config[:db_statement] == :omit` and `unless config[:db_statement] == :obfuscate`.

Reply:
@robertlaurin you're not being too nitpicky - we definitely want to cache the results of obfuscation if we're going to obfuscate it later on. @arielvalentin and I discovered that the string-munging obfuscation we do in the mysql instrumentation has a performance penalty (and the pg implementation is quite similar). So if we only have to pay that once for obfuscation, that's much better.

It's a moot point now that the change was made differently, but I don't think you were being overly cautious at all!